src/toil/jobStores/aws/jobStore.py

```diff
@@ -307,7 +307,8 @@ def _extractKeyInfoFromUrl(url):
         """
         s3 = boto.connect_s3()
         bucket = s3.get_bucket(url.netloc)
-        key = bucket.new_key(url.path[1:])
+        key = bucket.get_key(url.path[1:])
+        key = key if key is not None else bucket.new_key(url.path[1:])
```

Comment:
I'm surprised that there are that many. Let's try it. This method is when we first learn whether bucket and key exist. If bucket doesn't exist we can fail right away. If the key doesn't exists we should return (bucket, None). Methods that rely on the key to exist should test for None. Methods that write to the key, and hence need a Key object, should use new_key explicitly. I can only think of one place where that is necessary.